test/e2e/framework/volume/fixtures.go

```diff
@@ -486,8 +486,12 @@ func TestVolumeClient(f *framework.Framework, config TestConfig, fsGroup *int64,
 	clientPod, err := runVolumeTesterPod(f.ClientSet, config, "client", false, fsGroup, tests)
 	if err != nil {
 		framework.Failf("Failed to create client pod: %v", err)
-
 	}
+	defer func() {
```

Comment:
looks like some callers already do some sort of cleanup (and some dont): 
https://github.com/kubernetes/kubernetes/blob/574acbe31025bd7a298eae5dc72b0256a7a9b11a/test/e2e/storage/flexvolume.go#L68

Looks like these cleanups are overlapping. Could you  remove the redundant ones?

Reply:
Removed client cleanup from TestCleanup. Therefore I renamed it to TestServerCleanup. And  removed several calls to TestCleanup from tests that don't use a server.